### DISCUSSION TITLE
Borderless window with native buttons

### DISCUSSION THREAD

**Discussion by WrobotGames** [godotengine/godot-proposals#4132] [Platforms]
Currently we have two 'modes' for windows: Normal (With system buttons, title and icon) and borderless. 

Sometimes you want the space that the title bar occupies, but still want the system buttons with their behaviour (e.g. windows 11 maximise button hover, visible below). 
![image](https://user-images.githubusercontent.com/66079814/155899615-50502368-159e-4146-bb33-3af2f65d6d51.png)
_Word is borderless, but with the system buttons and native window 'edge'_
This should be implemented on Linux, Windows, and MacOS. On windows this can be done with 'Win32 DWM'

One thing to watch out for is that some os's have the buttons on the left side. So people using this feature should check the operating system and make sure there is no important ui under the buttons.

One step further than just extending the content would be drawing on top of the os title bar. (like chrome does), but this is a lot more work to implement and doesnt add much more than theming, so I wouldnt spend time on this. (example) 
![image](https://user-images.githubusercontent.com/66079814/155900441-4cf35b53-f929-4503-a323-cb696d68327c.png)

**Comment by Calinou** [comment]
See also https://github.com/godotengine/godot-proposals/issues/1868, which is tangentially related. Implementing client-side decorations is planned in the long run (as it would also allow for things like https://github.com/godotengine/godot/pull/34528 to be implemented), but there's no ETA for it.

**Comment by Calinou** [comment]
Closing in favor of https://github.com/godotengine/godot-proposals/issues/4846. This is now implemented on macOS; support for other platforms requires changes to the platform code to make client-side decorations possible.
